lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -94,29 +99,251 @@ class IntProgCmdProvider: public CmdProvider {
         }
     }
 
+/**
+ * Get the command type.
+ * @return The command type, which is INIT for initialization.
+ */
+CMD_TYPE getCmdType() const override 
+{
+    return INIT;
+}
+
 private:
-    static const uint32_t SEQ_LENGHT = 12; /**< Length of the command sequence */
+    static const uint32_t SEQ_LENGHT = 12; /**< Number of required commands to initilize the Flash.*/
     uint32_t m_index; /**< current index in sequence */
-    static const CommandInfo m_cmds[SEQ_LENGHT]; /**< Array of command information */
-    static const ResponseInfo m_responses[SEQ_LENGHT]; /**< Array of response information */
+    static  CommandInfo m_cmds[SEQ_LENGHT]; /**< Array of command information */
+    static  ResponseInfo m_responses[SEQ_LENGHT]; /**< Array of response information */
 };
 
-#if 0
+/**
+ * @class FwProgCmdProvider
+ * @brief Implementation of the command provider for firmware programming.
+ *
+ * This class provides commands for firmware programming, specifically tailored
+ * for the Zumo32U4 device.
+ */
 class FwProgCmdProvider : public CmdProvider {
     public:
-        FwProgCmdProvider(FileHandle * fs);
+        /**
+         * @brief Constructor for the FwProgCmdProvider class.
+         * @param fileName The name of the file to be used for firmware programming.
+         * Initializes a new instance of the FwProgCmdProvider class with the provided file name.
+         * @remarks The file specified by fileName should contain the firmware to be programmed.
+         */
+        FwProgCmdProvider(const char* fileName):
+            m_fileName(fileName),
+            m_updatedCmd({nullptr, 0}), 
+            m_updatedCmdBuffer(nullptr),
+            m_updatedCmdBufferSize(0),
+            m_currWriteMemAddr(0x0000)
+        {}
+
+    /**
+     * @brief Resets the command provider to its initial state.
+     *
+     * This method resets the internal index that tracks the current command
+     * in the command sequence memory.
+     */
+    virtual void reset() {/*Intentionally empty.*/}
+
+    /**
+     * @brief Retrieves the next command and the next response.
+     *
+     * This method returns the next command and its associated response
+     * from the command sequence memory.
+     *
+     * @param cmd Reference to a pointer to the next command.
+     * @param rsp Reference to a pointer to the next response.
+     * @return True if there is a next command and next response, false otherwise.
+     */
+    virtual bool next(const CommandInfo*& cmd, const ResponseInfo*& rsp)
+    {
+        /*Keep original command.*/
+        const CommandInfo* originalCmd = &m_cmds[nextCmd];
+
+        /*Copy original command.*/
+        CommandInfo updatedCmd = *originalCmd;

```

Comment:
not needed as having no value add. Use m_updateCmd directly below.

Reply:
It will be take care by the next commit.